scipy/stats/tests/test_stats.py

```diff
@@ -7917,6 +7917,222 @@ def test_alias(self):
+        assert_equal(public, test_val)
+
+    def test_R_ci_quantile(self):
+        # Using the same test cases as the ci_quantile R function
+        # https://github.com/mayer79/confintr/blob/main/tests/testthat/test-ci_measures_of_location.R#L95
+
+        x = np.exp(np.arange(0, 1.01, 0.01))
+
+        lb, ub = stats.quantile_test(x, 
+                                     0, 
+                                     0.4, 
+                                     alternative='two-sided'
+                                     ).confidence_interval(0.95)
+        expected = [1.349859, 1.648721]
+        assert_almost_equal(expected, [lb,ub], decimal=5)
+
+        lb, ub = stats.quantile_test(x, 
+                                     0, 
+                                     0.5, 
+                                     alternative='two-sided'
+                                     ).confidence_interval(0.9)
+        expected = [1.506818, 1.803988]
+        assert_almost_equal(expected, [lb,ub], decimal=5)
+
+    def test_pval_ci_match(self):
+        # Verify that the following statement hold:
+        #
+        # The 95% confidence interval corresponding with alternative='less' 
+        # has -inf as its lower bound, and if xu is the upper bound, then xu 
+        # is the smallest element from the sample x such that:
+        # `stats.quantile_test(x, q=xu, p=p, alternative='less').pvalue``
+        # will be less than 5%.
+        # 
+        # And the corresponding statement for the alternative='greater' case.
+
+        # This test may fail if we the CI bound falls on a value present 
+        # multiple times in the sample. We make sure this does not happen by 
+        # generating samples via permutations rather than pure random.
+
+        for xi in x:
+            # alternative='less'
+            res = stats.quantile_test(xi, alternative='less')
+            lb, ub = res.confidence_interval()
+            if not np.isnan(ub):
+                i = np.where(xi == ub)[0][0]
+                assert_equal(
+                    stats.quantile_test(xi, 
+                                        q=xi[i], 
+                                        alternative='less'
+                                        ).pvalue <= 0.05, 
+                    True)
+                assert_equal(
+                    stats.quantile_test(xi, 
+                                        q=xi[i-1], 
+                                        alternative='less'
+                                        ).pvalue > 0.05, 
+                    True)
+            # alternative='greater'
+            res = stats.quantile_test(xi, alternative='greater')
+            lb, ub = res.confidence_interval()  
+            if not np.isnan(lb):
+                i = np.where(xi == lb)[0][0]
+                assert_equal(
+                    stats.quantile_test(xi, 
+                                        q=xi[i], 
+                                        alternative='greater'
```

Comment:
AFAICT, the only difference between this and the previous part of the test is the `alternative` and whether the index is incremented or decremented by one. Please `parameterize` the test for concision.

Reply:
okay, I'll look into how to do that

Reply:
Please see my PR to your branch.